src/algebra/order/field.lean

```diff
@@ -725,4 +726,34 @@ lemma one_div_pow_mono (a1 : 1 ≤ a) : monotone (λ n : ℕ, order_dual.to_dual
 lemma one_div_pow_strict_mono (a1 : 1 < a) : strict_mono (λ n : ℕ, order_dual.to_dual 1 / a ^ n) :=
 λ m n, one_div_pow_lt_one_div_pow_of_lt a1
 
+/-! ### Results about `is_lub` and `is_glb` -/
+
+lemma is_lub.mul_left {s : set α} (hc : 0 ≤ a) (hs : is_lub s b) :
+  is_lub ((λ b, a * b) '' s) (a * b) :=
+begin
+  rcases lt_or_eq_of_le hc with hc | rfl,
+  { exact (order_iso.mul_left₀ _ hc).is_lub_image'.2 hs, },
+  { simp_rw zero_mul,
+    rw hs.nonempty.image_const,
+    exact is_lub_singleton },
+end
+
+lemma is_lub.mul_right {s : set α} (hc : 0 ≤ a) (hs : is_lub s b) :
+  is_lub ((λ b, b * a) '' s) (b * a) :=
+by simpa [mul_comm] using hs.mul_left hc
+
+lemma is_glb.mul_left {s : set α} (hc : 0 ≤ a) (hs : is_glb s b) :
+  is_glb ((λ b, a * b) '' s) (a * b) :=
```

Comment:
Last comment; is this true as an iff (from `hs`)

Reply:
Not quite, the converse would require `0 < a`.

Reply:
Right, and in that case it's just a direct application of `is_lub_image` so not lemma worthy